Parsers/ASimWebSession/Parsers/ASimWebSessionF5ASM.yaml

```diff
@@ -0,0 +1,169 @@
+ParserName: ASimWebSessionF5ASM
+EquivalentBuiltInParser: _ASim_WebSession_F5ASM
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let DvcActionLookup = datatable (DeviceAction: string, DvcAction: string)
+  [
+      "Blocked", "Deny",
+      "blocked", "Deny",
+      "Passed", "Allow",
+      "passed", "Allow",
+      "Alerted", "Deny",
+      "alerted", "Deny"
+  ];
+  let EventSeverityLookup = datatable (LogSeverity: string, EventSeverity: string)
+  [
+      "0", "Low",
+      "1", "Low",
+      "2", "Low",
+      "3", "Low",
+      "4", "Medium",
+      "5", "Medium",
+      "6", "Medium",
+      "7", "High",
+      "8", "High",
+      "9", "High",
+      "10", "High"
+  ];
+  let parser=(disabled: bool=false) {
+      let AllData = CommonSecurityLog
+          | where DeviceVendor == "F5" and DeviceProduct == "ASM"
+          | invoke _ASIM_ResolveDvcFQDN('DeviceName')
+          | project-rename DvcIpAddr = DeviceAddress;
+      let GeneralEnforcementData = AllData
+          | where ((substring(DeviceEventClassID, 0, 1) == "2" and strlen(DeviceEventClassID) == 9) or DeviceEventClassID == Activity) and DeviceEventClassID !in ("Brute Force Attack", "IP Enforcer Attack", "Web Scraping Attack", "DoS Attack")
+          | parse-kv DeviceCustomString3 as (Host: string, ["User-Agent"]: string, Cookie: string, Referer: string) with (pair_delimiter="\\r\\n", kv_delimiter=":")
+          | parse DeviceCustomString3 with * "HTTP/" HttpVersion: string "\\r\\n" rest: string
+          | extend
+              EventResultDetails = tostring(FieldDeviceCustomNumber1)
+          | project-rename 
+              DstIpAddr = DestinationIP,
+              DstPortNumber = DestinationPort,
+              EventOriginalUid = ExtID,
+              HttpRequestMethod = RequestMethod,
+              NetworkApplicationProtocol = ApplicationProtocol,
+              HttpCookie = Cookie,
+              HttpHost = Host,
+              HttpReferrer = Referer,
+              HttpUserAgent = ['User-Agent'],
+              HttpRequestXff = DeviceCustomString5
+          | extend
+              EventResult = iff(toint(EventResultDetails) >= 400 or DeviceAction =~ "blocked", "Failure", "Success"),
+              HttpStatusCode = EventResultDetails,
+              AdditionalFields = bag_pack(
+                        "Full Request", DeviceCustomString3, 
+                        "Policy Name", DeviceCustomString1,
+                        "Attack Type", DeviceCustomString4,
+                        "Policy Apply Date", DeviceCustomDate1
+                    );
+      let AnomalyDetectionData = AllData
+          | where DeviceEventClassID in ("Brute Force Attack", "IP Enforcer Attack", "Web Scraping Attack", "DoS Attack")
+          | extend
+              EventResult = iff(DeviceAction =~ "passed", "Success", "Failure"),
+              DstIpAddr = DvcIpAddr,
```

Comment:
why this? Is it always the case?
DstIpAddr is destination site address. DvcIp is of system which reported the event

Reply:
As for Anomalydetectiondata, in sample data, we don't have field related to destination so we have mapped dst with dvc as Source fields are available separately.

Reply:
same as previous comment related to DstHostname ?

Reply:
As Destination is compulsory and we have source and device-related fields available from F5, we have mapped device fields with Destination.

Reply:
@jayeshprajapaticrest @niralishah-crest 
I didn't get the reason. DstHostname is not a mandatory field, you can skip it if not coming from product.

Reply:
DstIpAddr is not a mandatory field but Dst is a mandatory field and other than IpAddr we don't have such a field to map with Dst. Hence we have done mapping like this.

Reply:
@niralishah-crest @jayeshprajapaticrest 
If certain field is not available, we should not map any field to fulfil the mandatory requirements. Like in this case, I'm unable to understand if Dst is not coming, why we are mapping Dvc information just to fulfill the requirement? Are we sure Dst and Dvc refer to same device in this case? If yes, we are good, if not we should remove this mapping.

Reply:
@niralishah-crest if we don't have Dst for a certain type, pls don't map.